tests/suites/test_suite_fs_abstraction.function

```diff
@@ -0,0 +1,177 @@
+/* BEGIN_HEADER */
```

Comment:
A few tests that would be nice to have:

* Running out of memory when reading inputs. This case is buggy now so it would be a non-regression test.
* Seeking in a file that's more than 2^32-1 bytes long.
* `readdir` with a file name that doesn't fit in the buffer.
* Running into the limit on the number of open files.
* Failing read/write (may be difficult to arrange).
* Memory leak test: do a lot of operations on files and directories (open, read, maybe other stuff, close, repeat) and check that memory doesn't keep rising (both on the target and on the host).

Reply:
* Running out of memory when reading inputs. This case is buggy now so it would be a non-regression test.
---- This will require a framework for testing the frontend separately. Via mbed TLS test suites it would difficult to mock ```mbedtls_calloc```.
* Seeking in a file that's more than 2^32-1 bytes long.
---- It will fail certainly. For completeness I can send send & recv seek/ftell offset in two 32 bit transactions. However, I don't see how that will help us since on target we can't read in that big files and practically we would not be dealing with that big files.
* `readdir` with a file name that doesn't fit in the buffer.
---- This can be tested.
* Running into the limit on the number of open files.
---- This can be tested.
* Failing read/write (may be difficult to arrange).
---- This can be partially tested by trying write on a read only file.
* Memory leak test: do a lot of operations on files and directories (open, read, maybe other stuff, close, repeat) and check that memory doesn't keep rising (both on the target and on the host).
---- Again this could be an isolated test for ```frontend.c```. ```fsio.c``` and ```serialize.c``` work on caller's buffer and do not allocate any memory of their own.